File: src/core/checks/dangerous-tools.ts
import { BaseCheck, SkillContext, CheckResult } from '../types';

export class DangerousToolsCheck implements BaseCheck {
  id = 'dangerous-tools';
  name = 'Dangerous Tools Audit';
  weight = 1.0;

  async run(context: SkillContext): Promise<CheckResult> {
    const dangerous = ['run_shell_command', 'write_file', 'delete_file'];
    const found = context.tools.filter(t => dangerous.includes(t));

    if (found.length > 0) {
      return {
        id: this.id,
        name: this.name,
        score: 0,
        level: 'Critical',
        justification: `Skill requests highly dangerous tools: ${found.join(', ')}`
      };
    }

    return {
      id: this.id,
      name: this.name,
      score: 10,
      level: 'Low',
      justification: 'No dangerous tools detected.'
    };
  }
}

File: internal/checks/dangerous_tools.go
package checks

import (
	"context"
	"fmt"
	"strings"
	"github.com/albertowar/skillauditai/pkg/api"
	"github.com/albertowar/skillauditai/internal/behavioral"
	"github.com/albertowar/skillauditai/internal/provider"
)

type DangerousToolsCheck struct{}

func (c *DangerousToolsCheck) ID() string      { return "dangerous-tools" }
func (c *DangerousToolsCheck) Name() string    { return "Dangerous Tools Audit" }
func (c *DangerousToolsCheck) Weight() float64 { return 1.0 }

func (c *DangerousToolsCheck) Run(ctx context.Context, skill api.SkillContext, b *behavioral.Service) (api.CheckResult, error) {
	p := provider.Get(skill.Provider)
	dangerousList := p.DangerousTools()

	dangerousMap := make(map[string]bool)
	for _, dt := range dangerousList {
		dangerousMap[strings.ToLower(dt)] = true
	}

	var found []string
	for _, t := range skill.Tools {
		lowerTool := strings.ToLower(t)
		if dangerousMap[lowerTool] {
			found = append(found, t)
		}
	}

	if len(found) > 0 {
		return api.CheckResult{
			ID:            c.ID(),
			Name:          c.Name(),
			Score:         0,
			Level:         api.Critical,
			Justification: fmt.Sprintf("Skill requests highly dangerous tools: %s", strings.Join(found, ", ")),
		}, nil
	}

	return api.CheckResult{
		ID:            c.ID(),
		Name:          c.Name(),
		Score:         10,
		Level:         api.Low,
		Justification: "No dangerous tools detected.",
	}, nil
}
